dGame/dBehaviors/BehaviorContext.cpp

```diff
@@ -260,16 +260,32 @@ bool BehaviorContext::CalculateUpdate(const float deltaTime) {
 	return any;
 }
 
-void BehaviorContext::Interrupt() {
+void BehaviorContext::Interrupt(bool interruptAttack, bool interruptBlock, bool interruptCharge) {
 	std::vector<BehaviorSyncEntry> keptSync{};
-
 	for (const auto& entry : this->syncEntries) {
-		if (!entry.ignoreInterrupts) continue;
+		if (!entry.ignoreInterrupts) {
+			if (interruptAttack && entry.behavior->m_templateId == BehaviorTemplates::BEHAVIOR_ATTACK_DELAY) {
```

Comment:
could these be combined into one if statement?

Reply:
It could, but it would be quite large so keeping them as their own statement so its clear to a reader what is going on I feel would be more ideal